### ISSUE TITLE
@SuppressWarnings on field is not respected with Groovy 3

### ISSUE THREAD

**Issue by pbielicki** [CodeNarc/CodeNarc#617]
The following class:

```groovy
class Test {
    @Deprecated
    @SuppressWarnings("UnusedPrivateField")
    private String ignored
}
```

fails CodeNarc when used with Groovy 3 (see the reproducer project: https://github.com/britter/gradle-sandbox/tree/main/codenarc-suppress-annotation):

![image](https://user-images.githubusercontent.com/1133099/118253024-bbe8c900-b4a9-11eb-9c7a-16aa683f0590.png)

Here's the build scan: https://scans.gradle.com/s/dtwca744s3n4q

Field's `@SuppressWarnings` annotation is not respected. The workaround is to annotate the class or to use `// codenarc-disable-line` line comment.

It works fine with Groovy 2: https://scans.gradle.com/s/lkw2ng6sve4ek/dependencies?toggled=W1swXSxbMCwwXSxbMCwwLFswXV1d

**Comment by chrismair** [comment]
This looks like a Groovy bug. I opened an issue on Groovy:

[GROOVY-10181](https://issues.apache.org/jira/browse/GROOVY-10181): _Groovy 3 AST for annotated fields reports wrong lastLineNumber if field not initialized_.

A workaround is to add an initial value to the field causing the problem.

**Comment by chrismair** [comment]
This was fixed by Groovy 3.0.9.
